spatialmath/geometry_utils.go

```diff
@@ -0,0 +1,147 @@
+package spatialmath
+
+import (
+	"github.com/golang/geo/r3"
+)
+
+const floatEpsilon = 1e-6
+
+// This file incorporates work covered by the Brax project -- https://github.com/google/brax/blob/main/LICENSE.
+// Copyright 2021 The Brax Authors, which is licensed under the Apache License Version 2.0 (the “License”).
+// You may obtain a copy of the license at http://www.apache.org/licenses/LICENSE-2.0.
+
+// DistToLineSegment takes a line segment defined by pt1 and pt2, plus some query point, and returns the cartesian distance from the
+// query point to the closest point on the line segment.
+func DistToLineSegment(pt1, pt2, query r3.Vector) float64 {
```

Comment:
A lot of the code in this file feels familiar to me.  Is there a chance it is duplicating other code?

Reply:
This was meant to be moved over from motionplan/constraints.go, but it appears I forgot to actually commit that change.